code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,668 @@
 
 #undef ACTIVE_SETUPFIELDS
 #undef ACTIVE_HASFIELDS
+
+//Modular Shield Generator Start
+/obj/machinery/modular_shield_generator
```

Comment:
Please put this in its own file

Reply:
ill just name it modular_shield_gen.dm if thats not too long